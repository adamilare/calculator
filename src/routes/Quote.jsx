import { useEffect, useState } from 'react';
import getQuote from '../components/servises/quotes.service';

const Quote = () => {
  const [quote, setQuote] = useState({});
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(false);

  useEffect(() => {
    const apiData = async () => {
      const quoteResponse = await getQuote();
      if (!quoteResponse || quoteResponse.error) {
        setError(true);
        setLoading(false);
        return;
      }

      setError(false);
      setLoading(false);
      setQuote(quoteResponse.data[0]);
    };
    apiData();
  }, [setQuote]);

  return (
    <div className="quote-grp">
      {loading && <p>Loading...</p>}
      {error && <p>Something went wrong</p>}
      {quote && (
        <p>
          {quote.quote}
          <br />
          <span className="author">{quote.author}</span>
        </p>
      )}
    </div>
  );
};

export default Quote;
