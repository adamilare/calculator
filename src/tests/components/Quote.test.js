/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import axios from 'axios';
import Quote from '../../routes/Quote'; // eslint-disable-line

jest.mock('axios', () => ({
  get: jest.fn(() =>
    Promise.resolve({ data: [{ quote: 'Test quote', author: 'Test author' }] })
  ),
}));

test('Quote page displays correctly', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});
