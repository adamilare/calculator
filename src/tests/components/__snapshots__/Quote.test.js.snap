// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote page displays correctly 1`] = `
<div
  className="quote-grp"
>
  <p>
    Loading...
  </p>
  <p>
    <br />
    <span
      className="author"
    />
  </p>
</div>
`;
